### PR TITLE
Only set disks address type in the virtio case (bsc#987584)

### PR DESCRIPTION
**Pull request by dirkmueller**
For Xen/HyperV the PCI address is not settable but only an IDE
bus id would be setable (but apparently we don't care about that)
https://bugzilla.suse.com/show_bug.cgi?id=987584

### CODE REVIEW

**Comment by aplanas** [comment]
+1

**Comment by toabctl** [comment]
+q but let's wait for @bmwiedemann

File: scripts/lib/libvirt/libvirt_setup.py [hunk 1]
```diff
@@ -166,7 +166,8 @@ def compute_config(args, cpu_flags=cpuflags(), machine=None):
                 args.nodecounter,
                 i),
             'target_dev': targetdevprefix + ''.join(alldevices.next()),
-            'target_slot': '0x1{0}'.format(hex(i+9)[2:]),
+            'target_address':
+                "<address type='pci' slot='0x1{0}'/>".format(hex(i+9)[2:]),
```
Comment (bmwiedemann) [inline scripts/lib/libvirt/libvirt_setup.py]: wouldnt this still break xen?

File: scripts/lib/libvirt/libvirt_setup.py [hunk 2]
```diff
@@ -185,7 +186,8 @@ def compute_config(args, cpu_flags=cpuflags(), machine=None):
                     args.nodecounter,
                     i),
                 'target_dev': targetdevprefix + ''.join(alldevices.next()),
-                'target_slot': '0x1{0}'.format(i),
+                'target_address':
+                    "<address type='pci' slot='0x1{0}'/>".format(i),
```
Comment (bmwiedemann) [inline scripts/lib/libvirt/libvirt_setup.py]: dito

**Comment by jdsn** [comment]
:+1: lgtm - is it tested? (PR testing does not cover these changes)

**Pull request merged**